### DISCUSSION TITLE
Meaning of weights in community detection & centrality metrics

### DISCUSSION THREAD

**Discussion by omri-al** [networkx/networkx#7200] [Q&A]
I am building a directed graph with weights, where the higher the weight, the more "connected" the nodes are (in my use case weight is the sum of interactions between 2 nodes). I am using Networkx to model the graph and to compute 2 things:

communities
most influential node in each community for this, I use louvain_communities for communities detection, and multiple centrality metrics, which I compare (closeness_centrality, betweenness_centrality, eigenvector_centrality, pagerank)

**My question is**: For each of these metrics\algorithms (i.e., louvain and all of the centrality metrics), is the weight means "distance" or "closeness"? Should I use weight=weight or weight=1/weight?

I tried to read the documentation but its quite confusing

**Comment by dschult** [comment]
Some algorithms and some centrality measures use edge weights to mean "distance" and others use it to mean "closeness". We have followed what the literature/papers define it to be. For example, methods that use shortest_path measures to find the relationship between nodes tend to use edge weight as the distance. While measures that use "closeness" (like closeness_centrality) or community structure often use edge weight to measure the strength of the connection between nodes. So a larger value means a stronger connection.

Ideally the documentation should indicate what the edge weight attribute measures and from that you can tell well bigger values are more connected or less connected. Can you point out places/functions for which that is confusing?

Thanks!

**Reply by omri-al** [reply]
Thanks For your response.
I will write what I think about each of these functions, and please correct me if I'm wrong:

1. louvain_communities : closeness (the higher the weight, the more "connected" the nodes are)
2. closeness_centrality: I'm not sure, since it's about "closeness", but the parameter is "distance". so I'n that case how should I pass the weights dictionary?
3. betweenness_centrality: distance (the higher the weight, the more "distant" the nodes are)
4. eigenvector_centrality: closeness
5. pagerank: closeness (here it's not very clear)
6. in_degree: closeness

**Reply by omri-al** [reply]
@dschult
